x-pack/plugins/maps/public/routing/routes/maps_app/index.ts

```diff
@@ -50,9 +59,16 @@ function mapStateToProps(state = {}) {
   };
 }
 
-function mapDispatchToProps(dispatch) {
+interface DispatchSetQueryArgs {
+  filters?: Filter[];
+  query?: Query;
+  timeFilters?: TimeRange;
+  forceRefresh?: boolean;
+}
+
+function mapDispatchToProps(dispatch: ThunkDispatch<Promise<string>, MapStoreState, any>) {
```

Comment:
Looking at [discuss issue](https://github.com/reduxjs/redux-thunk/issues/231) for `ThunkDispatch`, it appears that ThunkDispatch is generically typed like `ThunkDispatch<S, E, A>` where S is store state, E is extra argument, and A is the action. Should this be `ThunkDispatch<MapStoreState, void, AnyAction>` instead?

Reply:
Yes, you are right. I have made the changes.